src/Http/samples/MinimalSampleFSharp/Program.fs

```diff
@@ -0,0 +1,20 @@
+open System
+open Microsoft.AspNetCore.Builder
+open Microsoft.Extensions.Hosting
+
+[<EntryPoint>]
+let main args =
+    let app = WebApplication.Create(args);
+
+    if app.Environment.IsDevelopment() then
+        app.UseDeveloperExceptionPage() |> ignore
+
+    app.MapGet("/", Func<string>(fun () -> "Hello World!")) |> ignore
+
+    app.MapGet("/plaintext", Func<string>(fun () -> "Hello, World!")) |> ignore
+    app.MapGet("/json", Func<obj>(fun () -> {| message = "Hello, World!" |} :> obj)) |> ignore
```

Comment:
Hmm, weird that the cast to `obj` is needed here. F# anonymous records compile down to anonymous objects for interop

Reply:
If it's needed, then this might be cleaner:

`upcast {| message = "Hello, World!" |} `

Reply:
You can also try `Func<_>` and see if no cast is necessary